app/proxy.js

```diff
@@ -70,6 +70,15 @@ function authenticate(context, callback, shouldNotCatch) {
         if (site.forwardedLoginHeader && context.login) {
           context.options.headers[site.forwardedLoginHeader] = context.login;
         }
+        if (site.forwardedLoginSecret && context.login) {
+          var addedHeaders=require('couch-proxy-auth')(context.login,site.forwardedLoginRoles || 'protect',site.forwardedLoginSecret);
+          for (var newHeader in addedHeaders) {
+            // since roles are forced to a value, the role header is given by couch-proxy-auth
+            // so can't be forged
+            //removeHeader(context.options.headers,newHeader);
```

Comment:
Oops, I made a test, forgot t switch back. It seems that if the same header is present twice, last value is taken in consideration, so commenting the removeHeader didn't change the result *in this case*. But for full security, I will put it back.